lib/engine/route.rb

```diff
@@ -205,6 +205,17 @@ def check_overlap!
       end
     end
 
+    def check_edge_reuse!
+      used_edges = paths.flat_map do |path|
+        path.exits.map { |edge| [path.hex, edge] }
+      end
+      dupes = used_edges.group_by(&:itself).select { |_k, v| v.length > 1 }
```

Comment:
always use size, v.size

Reply:
this can be simplified as

hexes = paths
  .flat_map { |path| path.exits.map ...}
  .group_by(&:itself)
  .select { |_, v| v.size > 1 }
  .map { |k, v| k.first.name }
  .sort
  .join(',')
  
@game.game_error('') if hexes.any?